app/src/androidTest/java/com/ncnf/views/activities/group/FriendsTrackerActivityTest.java

```diff
@@ -44,54 +60,92 @@
-import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.*;
 import static org.mockito.Mockito.when;
 
 @HiltAndroidTest
 @UninstallModules(CurrentUserModule.class)
 public class FriendsTrackerActivityTest {
 
+    static private final User user1 = Mockito.mock(User.class);
+    static private final User user2 = Mockito.mock(User.class);
+    static private final GroupRepository groupRepository1 = Mockito.mock(GroupRepository.class);
+    static private final UserRepository userRepository1 = Mockito.mock(UserRepository.class);
+
+    private static final List<String> users = Collections.singletonList("1");
+    static private UUID gUuid = UUID.randomUUID();
+    static Intent intent;
+    static {
+        intent = new Intent(ApplicationProvider.getApplicationContext(), FriendsTrackerActivity.class);
+        Bundle bundle = new Bundle();
+        bundle.putString("GROUP_ID", gUuid.toString());
+        intent.putExtras(bundle);
+    }
+
 
     private final HiltAndroidRule hiltRule = new HiltAndroidRule(this);
-    private GeoPoint p1;
+    private static GeoPoint p1 = new GeoPoint(0.03, 0.03);
 
+
+    static private final Group g = Mockito.mock(Group.class);
     @Rule
     public final ServiceTestRule mServiceRule = new ServiceTestRule();
 
 
     @InjectMocks
-    private final ActivityScenarioRule scenario = new ActivityScenarioRule<>(FriendsTrackerActivity.class);
+    private final ActivityScenarioRule scenario = new ActivityScenarioRule<>(intent);
 
     @Rule
     public RuleChain testRule = RuleChain.outerRule(hiltRule).around(scenario);
 
     @BindValue
-    public User user = mock(User.class);
+    public User user = user1;
+
+    @BindValue
+    public GroupRepository groupRepository = groupRepository1;
 
     @BindValue
-    public FirebaseDatabase dbs = mock(FirebaseDatabase.class);
+    public UserRepository userRepository = userRepository1;
+
+
+    @BeforeClass
+    static public void injects() {
+        when(user1.loadUserFromDB()).thenReturn(CompletableFuture.completedFuture(user1));
+        when(user1.getUuid()).thenReturn("0");
+        when(user2.getUuid()).thenReturn("1");
+        ArrayList<Group> l = new ArrayList<>();
+        l.add(g);
+
+        ArrayList<String> s = new ArrayList<>();
+        s.add("1");
+        when(g.getAttendees()).thenReturn(s);
+
+        when(groupRepository1.loadGroup(gUuid.toString())).thenReturn(CompletableFuture.completedFuture(g));
+       // when(user1.getParticipatingGroupsIds().contains("0")).thenReturn(true);
+
+        when(userRepository1.getUserFullName("0")).thenReturn(CompletableFuture.completedFuture("Johm"));
+        when(userRepository1.getUserPosition("0")).thenReturn(CompletableFuture.completedFuture(new GeoPoint(0, 0)));
+
+        when(userRepository1.getUserFullName("1")).thenReturn(CompletableFuture.completedFuture("Taylor"));
+        when(userRepository1.getUserPosition("1")).thenReturn(CompletableFuture.completedFuture(p1));
```

Comment:
Using Mockito.when, if you want to match exact arguments, use the `eq` method:
`userRepository1.getUserFullName(eq("1"))`